core/trino-main/src/test/java/io/trino/server/security/oauth2/TestOAuth2WebUiAuthenticationFilterWithJwt.java

```diff
@@ -13,19 +13,37 @@
  */
 package io.trino.server.security.oauth2;
 
+import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
 import com.google.common.io.Resources;
+import com.google.inject.Key;
 import io.airlift.http.client.HttpClientConfig;
 import io.airlift.http.client.jetty.JettyHttpClient;
+import io.airlift.http.server.HttpServerInfo;
 import io.jsonwebtoken.Claims;
 import io.jsonwebtoken.Jws;
 import io.jsonwebtoken.Jwts;
+import io.trino.security.AccessControlManager;
+import io.trino.server.security.TestResourceSecurity.TestSystemAccessControl;
 import io.trino.server.security.jwt.JwkService;
 import io.trino.server.security.jwt.JwkSigningKeyResolver;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.testng.annotations.Test;
 
+import javax.ws.rs.core.HttpHeaders;
+
+import java.io.IOException;
 import java.net.URI;
+import java.util.function.Consumer;
 
+import static io.airlift.http.client.HttpUriBuilder.uriBuilderFrom;
+import static io.trino.server.security.TestResourceSecurity.TestSystemAccessControl.NO_IMPERSONATION;
+import static io.trino.server.security.TestResourceSecurity.TestSystemAccessControl.WITH_IMPERSONATION;
+import static javax.ws.rs.core.Response.Status.OK;
+import static javax.ws.rs.core.Response.Status.UNAUTHORIZED;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.assertEquals;
 
 public class TestOAuth2WebUiAuthenticationFilterWithJwt
```

Comment:
Shouldn't this be single-threaded now? The access control is being modified by one of the tests.

Reply:
Good point 👍